src/client/csfle/state_machine.rs

```diff
@@ -89,11 +113,22 @@ impl CryptExecutor {
                         Error::internal("db required for NeedMongoMarkings state")
                     })?;
                     let op = RawOutput(RunCommand::new_raw(db.to_string(), command, None, None)?);
-                    let mongocryptd_client = self
-                        .mongocryptd_client
+                    let mongocryptd = self
+                        .mongocryptd
                         .as_ref()
                         .ok_or_else(|| Error::internal("mongocryptd client not found"))?;
-                    let response = mongocryptd_client.execute_operation(op, None).await?;
+                    let mut mongocryptd_client = mongocryptd.client.lock().await;
+                    let mut result = mongocryptd_client.execute_operation(op.clone(), None).await;
+                    if let Err(e) = &result {
+                        if matches!(&*e.kind, ErrorKind::ServerSelection { .. }) {
+                            *mongocryptd_client = mongocryptd.opts.spawn().await?;
```

Comment:
I don't think we need to recreate the `Client` each time we spawn mongocryptd, we just need to restart the process and the monitor in the client should rediscover the new process. This will also allow us to drop the `Mutex`, which is currently limiting us to one operation at a time using this client.

We will need a lock around something to ensure we don't try to spawn `mongocryptd` multiple times at once though, maybe around the opts?

Reply:
Updated - this meshed nicely with the switch to the runtime process APIs, since it turns out keeping a `Child` handle around is needed to have the opportunity to reap dead processes.